### ISSUE TITLE
Change Language -> Update UI to Destination-Language / HotKey for Voice-Assistant

### ISSUE THREAD

**Issue by MickLesk** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#142]
Hey Guy!

I think your Google Assistant for Windows is mega! I've been looking for something like this for a long time to operate my smarthome while working and without using my phone. (Except with my Nest devices). The app is really very nice!

1st feature: 
A wish would be but from me, if you set the language (in my case "German"), that the rest of the UI of the app also "translates".
(I can also be available to you as a "translator" :-) )

2nd Feature:
When changing the language, you should also be offered the dialog to restart, otherwise the language does not change correctly, you must first scroll down in the settings to "relaunch Assistant". 

3rd feature:
A key combination or similar to directly "start" the voice assistant (apart from "OK Google" which you "plan" in the next versions).

**Comment by Melvin-Abraham** [comment]
@MickLesk

> A wish would be but from me, if you set the language (in my case "German"), that the rest of the UI of the app also "translates".
> (I can also be available to you as a "translator" :-) )

This will be worked upon sometime soon, once all the essential features and bug fixes are addressed ;)

> When changing the language, you should also be offered the dialog to restart, otherwise the language does not change correctly, you must first scroll down in the settings to "relaunch Assistant".

The language seems to change just fine without requiring a reload. Can you elaborate on "language does not change correctly"?

> A key combination or similar to directly "start" the voice assistant (apart from "OK Google" which you "plan" in the next versions).

This feature is already baked in. You can enable **Settings** > **Conversation** > **Enable microphone on application startup**, so that whenever you launch the assistant with hotkey, the microphone starts immediately.

**Comment by MickLesk** [comment]
@Melvin-Abraham 
> The language seems to change just fine without requiring a reload. Can you elaborate on "language does not change correctly"?
> Hey,

it's not quite like that. I have now changed from "German" to "English (usa)", and the display for example "how will the weather today" is still "German", the "voice output" is also "German" but the values -> instead of 5°C are 36°F (linguistic but not written) output. 

This all works only after restart or relaunch.

> This feature is already baked in. You can enable **Settings** > **Conversation** > **Enable microphone on application startup**, so that whenever you launch the assistant with hotkey, the microphone starts immediately.

 And how do I configure the hotkey? I like to use the function keys F1-F12 for such things, in which case I would take F5.

**Comment by eqbirvin** [comment]
> > This feature is already baked in. You can enable **Settings** > **Conversation** > **Enable microphone on application startup**, so that whenever you launch the assistant with hotkey, the microphone starts immediately.
> 
> And how do I configure the hotkey? I like to use the function keys F1-F12 for such things, in which case I would take F5.

I was going to start a new ticket to ask about this but I really don't want to inidate you, mate. we all really appreciate the work you have done!

**Comment by MickLesk** [comment]
> > > This feature is already baked in. You can enable **Settings** > **Conversation** > **Enable microphone on application startup**, so that whenever you launch the assistant with hotkey, the microphone starts immediately.
> > 
> > 
> > And how do I configure the hotkey? I like to use the function keys F1-F12 for such things, in which case I would take F5.
> 
> I was going to start a new ticket to ask about this but I really don't want to inidate you, mate. we all really appreciate the work you have done!


i found the following in the description, but i would like to have my own hotkey configured. :-) But maybe it helps you for now. 

**Default Keyboard Shortcut**
Windows: Win + Shift + A
MacOS: Cmd + Shift + A
Linux: Super + Shift + A

**Comment by esau-morais** [comment]
@MickLesk, to set your own shortcut, you just need to open the settings and scroll down till find an option called "Assistant Hotkey" and it'll listen to your keyboard

**Comment by Melvin-Abraham** [comment]
@MickLesk

> it's not quite like that. I have now changed from "German" to "English (usa)", and the display for example "how will the weather today" is still "German", the "voice output" is also "German" but the values -> instead of 5°C are 36°F (linguistic but not written) output.

This setting does not apply language changes to queries that have already been asked but affects the subsequent ones.

> And how do I configure the hotkey?

This feature has already been implemented in the source just not available yet. Till then you can clone this repo and build this project on your own if you want to test...
